pkg/testutil/helpers.go

```diff
@@ -108,6 +111,27 @@ func DeleteAndWaitUntilNotFound(ctx context.Context, c *RecordingClient, obj run
 	return WaitUntilNotFound(ctx, c, obj, options...)
 }
 
+// UpdateObject uses Poll based approach to update object to avoid update failures due to resource version conflicts.
+func UpdateObject(ctx context.Context, c *RecordingClient, obj runtime.Object, updateFunc func() error) error {
```

Comment:
```suggestion
func TryUpdateUntil(ctx context.Context, c client.Client, obj runtime.Object, updateFunc func() error) error {
```

siliar to common mutex/CAS nominclature: http://www.cplusplus.com/reference/mutex/timed_mutex/